zarp/run/snakemake.py

```diff
@@ -1 +1,126 @@
 """Module for executing Snakemake workflows."""
+
+import subprocess
+from typing import List, Optional
+
+from zarp.config.models import (ExecModes, InitRun,
+                                DependencyEmbeddingStrategies)
+
+
+class SnakemakeExecutor:
+    """Run snakemake with system calls.
+
+    Args:
+        run (InitRun): Run-specific parameters.
+
+    Attributes:
+        run_dict (dict): Dictionary with run-specific parameters.
+        success (bool): Indicator for successful snakemake run.
+        run_list (list[str]): List containing strings for snakemake call.
+
+    Example:
+        The example below expects a valid `Snakefile` (e.g. `touch Snakefile`)
+        in the current working directory.
+        It constructs a run with default values and runs it.
+        >>> mysnk = SnakemakeExecutor(InitRun())
+        >>> mysnk.prepare_run(snkfile = "Snakefile")
+        >>> mysnk.run()
+        >>> assert mysnk.get_success()
+
+    """
+
+    def __init__(self, run: InitRun) -> None:
+        """Class constructor."""
+        self.run_dict = run.dict()
+        self.success: Optional[bool] = None
+        self.run_list: List[str] = []
+
+    def set_run_list(self, run_list: list) -> None:
```

Comment:
Here and elsewhere:

Getters and setters aren't really used in Python. And if people insist on using them, then the best way to use them is through properties. See here: https://stackoverflow.com/questions/2627002/whats-the-pythonic-way-to-use-getters-and-setters

The important part for "setter" methods is that they do not return anything, i.e., no method should set an attribute AND return something. Or, for that matter, set more than one attribute. I will rename the `prepare_run` method to `set_command` to make this clearer.

I will remove the getters and setters in the next commit.